### PR TITLE
[IIIF-446] Refactor how the DELEGATE_URL environment variable is passed to tests, revise Travis config to call these additional tests

### PR DESCRIPTION
**Pull request by hardyoyo**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,10 @@ language: ruby
 env:
   matrix:
   - "CANTALOUPE_VERSION=stable"
+  - "CANTALOUPE_VERSION=stable_with_delegate"
```
Comment (ksclarke) [inline .travis.yml]: Couldn't we just test for the presence of the DELEGATE_URL in the ENV? Why do we need a different version value? You can matrix values like:

- "CANTALOUPE_VERSION=stable" "DELEGATE_URL=http://a.url.here"
- "CANTALOUPE_VERSION=stable"
- "CANTALOUPE_VERSION=dev" "DELEGATE_URL=http://a.url.here"
- "CANTALOUPE_VERSION=dev"

File: spec/Dockerfile.erb [hunk 2]
```diff
@@ -2,9 +2,6 @@ FROM <%= reg_username %>cantaloupe_<%= cantaloupe_version %>
 
 USER root
 
-# this is only for testing purposes, it is not included in the dockerhub images
-ENV DELEGATE_URL = "https://raw.githubusercontent.com/UCLALibrary/cantaloupe-delegate/master/lib/delegates.rb"
```
Comment (ksclarke) [inline spec/Dockerfile.erb]: Rather than hard coding here we could pass in the travis ENV matrix and include it if it's set.

File: Rakefile [hunk 3]
```diff
@@ -15,6 +15,20 @@ task :test_stable do
   abort 'Stable spec failed' unless RSpec::Core::Runner.run(['spec/cantaloupe_spec.rb']).zero?
 end
 
+desc 'Test the stable version with a DELEGATE_URL'
```
Comment (ksclarke) [inline Rakefile]: We wouldn't need these with the matrix ENV option

File: spec/Dockerfile.erb (hunk 2 above)
Comment (hardyoyo) [reply spec/Dockerfile.erb]: Are you sure you're looking at the correct side of the diff? I agree, I deleted this.

File: .travis.yml (hunk 1 above)
Comment (hardyoyo) [reply .travis.yml]: The environment variable doesn't actually pass through, which is a natural assumption, I understand why you'd make it. We have to re-create the Docker ENV in the cantalupe_spec.rb file (scan on down a bit).

File: Rakefile (hunk 3 above)
Comment (hardyoyo) [reply Rakefile]: This is true for Travis, not true for running rake locally, unless I'm mistaken? Rake doesn't know about the Travis matrix, right?

Comment (ksclarke) [reply Rakefile]: A Rakefile is just a Ruby script. The Travis matrices are just ENVs. If you put `puts ENV['FRANK']` in the rubocop task and then run `FRANK=bob rake rubocop` you'll see "bob" output. The only reason we put the CANTALOUPE_VERSION in the Rakefile is to simplify our running both of those two options for dev purposes. I think we could rely on Travis to run the delegate script tests (unless we wanted to run locally, then we could run with DELEGATE_URL=url-here rake). Setting the matrix wouldn't affect our local runs, that's right, though.

I was just proposing a simplification, not saying what you had wouldn't work or anything.

File: spec/Dockerfile.erb (hunk 2 above)
Comment (ksclarke) [reply spec/Dockerfile.erb]: You're right. I was seeing that backwards.

File: .travis.yml (hunk 1 above)
Comment (ksclarke) [reply .travis.yml]: I should pass through. Travis matrices are just system ENVs. See below.

**Pull request merged**